File: sol/solution/0001-0100/0011. Container With Most Water/Solution.scala
object Solution {
    def maxArea(height: Array[Int]): Int = {
        var left = 0
        var right = height.length - 1
        var maxArea = 0
        while (left < right) {
            val width = right - left
            val currentHeight = math.min(height(left), height(right))
            maxArea = math.max(maxArea, width * currentHeight)
            if (height(left) < height(right)) {
                left += 1
            } else {
                right -= 1
            }
        }
        maxArea
    }
}

File: sol/solution/0001-0100/0011. Container With Most Water/Solution.dart
class Solution {
  int maxArea(List<int> height) {
    int left = 0, right = height.length - 1;
    int maxArea = 0;
    while (left < right) {
      int width = right - left;
      int currentHeight = height[left] < height[right] ? height[left] : height[right];
      maxArea = maxArea > (width * currentHeight) ? maxArea : (width * currentHeight);
      if (height[left] < height[right]) {
        left++;
      } else {
        right--;
      }
    }
    return maxArea;
  }
}